account_invoice_analytic_search/models/account_invoice.py

```diff
@@ -0,0 +1,33 @@
+# -*- coding: utf-8 -*-
+# Copyright 2014-2017 Eficent Business and IT Consulting Services S.L.
+#   (http://www.eficent.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+
+
+class AccountInvoice(models.Model):
+
+    _inherit = "account.invoice"
+
+    @api.multi
+    @api.depends('invoice_line_ids.account_analytic_id')
+    def _compute_analytic_accounts(self):
+        for invoice in self:
+            invoice.account_analytic_ids =\
+                invoice.mapped('invoice_line_ids.account_analytic_id.id')
+
+    @api.multi
+    def _search_analytic_accounts(self, operator, value):
+        invoice_line_obj = self.env['account.invoice.line']
+        invoice_line_ids = invoice_line_obj.search(
+            [('account_analytic_id', operator, value)])
+        return [('id', 'in', tuple(invoice_line_ids.mapped('invoice_id').ids))]
```

Comment:
@aheficent As I said in https://github.com/OCA/account-invoicing/pull/298, this is not sufficent as this can lead to almost three searches (invoice_line.search(), invoice_line_ids.mapped() and return result).

Do just :
```python
def  _search_analytic_accounts(self, operator, value):
    return [('invoice_line_ids.account_analytic_id', operator, value)]
```